pint/util.py

```diff
@@ -304,7 +302,7 @@ def pi_theorem(quantities: dict[str, Any], registry: UnitRegistry | None = None)
 
 def solve_dependencies(
     dependencies: dict[TH, set[TH]],
-) -> Generator[set[TH], None, None]:
+) -> Generator[set[TH]]:
```

Comment:
Change this to `Iterator[set[TH]]` from collection.abc

Reply:
My bad it's still valid

Reply:
I think they changed this in 3.10 or 3.11, it was not ok before.